packages/wallets/src/components/WalletAdd.tsx

```diff
@@ -61,17 +59,10 @@ export default function WalletAdd() {
 
     try {
       const fingerprint = await addKey({
-        mnemonic: words,
-        type: 'new_wallet',
+        mnemonic: words.join(' '),
+        ...(label && { label: label.trim() }),
```

Comment:
Surprisingly `{...""}` and `{...false}` works but I rather prefer
```javascript
label: label.trim() ? label.trim() : undefined,
```
because it is more clear to me.

Reply:
There is a subtle but important difference between:
`...(label && { label: label.trim() })`
and
`label: label.trim() ? label.trim() : undefined,`

In the first case, if label is `''` or `undefined`, it won't include the `label` field at all. In the second case, it will produce `label: undefined`

The backend behaves differently depending on the presence of `label` in the payload and will produce an error if label is an empty string or null.

Perhaps the backend should behave differently, or we could handle this at the packages/api layer, but I'm inclined to leave it as I have it now.

I'll add comments in both locations.

Reply:
OK, it makes sense.

I just thought the two are equivalent after putting values to `addPrivateKey` function.

![image](https://user-images.githubusercontent.com/84098616/200155175-23777e57-f5b2-4324-b63b-ac3e363ed326.png)

![image](https://user-images.githubusercontent.com/84098616/200155290-c6fc1bfc-032d-422f-b051-bf1e6fade279.png)